PWGJE/TableProducer/jettaggerhf.cxx

```diff
@@ -21,64 +22,113 @@
 
 #include "PWGJE/DataModel/Jet.h"
 #include "PWGJE/DataModel/JetTagging.h"
+#include "PWGJE/Core/FastJetUtilities.h"
 #include "PWGJE/Core/JetTaggingUtilities.h"
 #include "PWGJE/Core/JetDerivedDataUtilities.h"
 
 using namespace o2;
 using namespace o2::framework;
 using namespace o2::framework::expressions;
 
-template <typename JetTableData, typename JetTableMCD, typename JetTaggingTableData, typename JetTaggingTableMCD>
+template <typename JetTableData, typename JetTableMCD, typename JetTaggingTableData, typename ConstTaggingTableData, typename ConstTCTaggingTableData, typename JetTaggingTableMCD, typename ConstTaggingTableMCD, typename ConstTCTaggingTableMCD>
 struct JetTaggerHFTask {
 
   Produces<JetTaggingTableData> taggingTableData;
+  Produces<ConstTaggingTableData> constTableData;
+  Produces<ConstTCTaggingTableData> constTCTableData;
   Produces<JetTaggingTableMCD> taggingTableMCD;
+  Produces<ConstTaggingTableMCD> constTableMCD;
+  Produces<ConstTCTaggingTableMCD> constTCTableMCD;
 
+  Configurable<bool> doWShower{"doWShower", false, "find jet origin included gluon spliting"}; // true:: remove gluon spliting
   Configurable<bool> doAlgorithm1{"doAlgorithm1", false, "fill table for algoithm 1"};
   Configurable<bool> doAlgorithm2{"doAlgorithm2", false, "fill table for algoithm 2"};
   Configurable<bool> doAlgorithm3{"doAlgorithm3", false, "fill table for algoithm 3"};
   Configurable<float> maxDeltaR{"maxDeltaR", 0.25, "maximum distance of jet axis from flavour initiating parton"};
 
+  using JetTagTracksData = soa::Join<aod::JTracks, aod::JTrackTagDcas, aod::JTrackTagDcaCovs>;
+  using JetTagTracksMC = soa::Join<aod::JTracks, aod::JMcTrackLbs, aod::JTrackTagDcas, aod::JTrackTagDcaCovs>;
+
   void processDummy(aod::Collision const& collision)
   {
   }
   PROCESS_SWITCH(JetTaggerHFTask, processDummy, "Dummy process", true);
 
-  void processData(aod::JCollision const& collision, JetTableData const& jets, aod::JTracks const& tracks)
+  void processData(soa::Join<aod::JCollisions, aod::JCollisionPIs>::iterator const& jcollision, aod::Collisions&, JetTableData const& jets, aod::JTracks&, JetTagTracksData const& tracks)
```

Comment:
please here use the predefined "using: keyowrds found at the bottom of DataModel/Jet.h